sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientCodeSnippets.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.data.tables.implementation.models.TableServiceErrorException;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class TableClientCodeSnippets {
+
+    public static void methods() {
+
+        //create a tableServiceClient
+        TableServiceClient tableServiceClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildClient();
+
+        //create TableClient
+        TableClient tableClient = new TableClientBuilder()
+            .connectionString("connectionString")
+            .tableName("OfficeSupplies")
+            .buildClient();
+
+
+        //create a table
+        try {
+            tableServiceClient.createTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            //use azure core errors? based on
+            System.out.println("Create Table Unsuccessful. Error: " + e);
+        }
+
+        //delete  table
+        try {
+            tableServiceClient.deleteTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            System.out.println("Delete Table Unsuccessful. Error: " + e);
+        }
+
+        //query tables
+        String selectString = "$selectString= TableName eq 'OfficeSupplies'";
+
+        try {
+            List<AzureTable> responseTables = tableServiceClient.queryTables(selectString);
+        } catch (HttpResponseException e) {
+            System.out.println("Table Query Unsuccessful. Error: " + e);
+        }
+
+
+        //insert entity
+        String tableName = "OfficeSupplies";
+        String row = "crayola markers";
+        String partitionKey = "markers";
+        HashMap<String, Object> tableEntityProperties = new HashMap<>();
+        TableEntity tableEntity = new TableEntity(tableName, row, partitionKey, tableEntityProperties);
+        try {
+            tableEntity = tableClient.insertEntity(tableEntity);
```

Comment:
This is good. The service will update some properties on the entity such as the internal `ETag` value, so the customer should always save and reuse the return value, rather than the original `tableEntity`.

Actually that makes me think that it might be hard to explain that to customers. If they *don't* save this value and instead keep using the instance created on L57 to perform updates, that should fail because that instance doesn't have the ETag. This leads me to think that maybe our `insertEntity` call should just accept the row, partitionKey, and properties directly. That way the customer won't create a `TableEntity` instance that could get them into trouble later on.